Add test cases for Transkribus related methods

kolakachi opened the pull request:
Added a method testFilterValidLangsTranskribusEngine in the EngineBaseTest class.
Added test method for methods related to Transkribus defined in AppExtension.

Bug: T343003:

kolakachi committed:
Add test cases for Transkribus related methods

Added a method testFilterValidLangsTranskribusEngine
in the EngineBaseTest class.
Added test method for methods related to Transkribus
defined in AppExtension.

Bug: T343003:

kolakachi committed:
Add test cases for Transkribus related methods

Added a method testFilterValidLangsTranskribusEngine
in the EngineBaseTest class.
Added test method for methods related to Transkribus
defined in AppExtension.

Bug: T343003:

kolakachi committed:
Fix linting errors

samwilson commented on tests/Engine/EngineBaseTest.php:
```diff
@@ -109,6 +125,33 @@ public function testFilterValidLangsTesseractEngine(
+	 * @covers EngineBase::filterValidLangs for Transkribus Engine
+	 * @dataProvider provideTranskribusLangs
+	 */
+	public function testFilterValidLangsTranskribusEngine(
```
This method looks very similar to the above `testFilterValidLangsTesseractEngine()`, and could perhaps be combined with that and only have one data provider method (which could provide an engine name as well). What do you think?

kolakachi replied:
Both methods have been Combined.

samwilson commented on tests/Engine/EngineBaseTest.php:
```diff
@@ -153,4 +213,13 @@ public function testLangNames(): void {
+	public function testValidLineIds(): void {
+		static::assertNotEmpty( $this->transkribusEngine->getValidLineIds( false, false ), "Missing line IDs" );
+		static::assertNotEmpty( $this->transkribusEngine->getValidLineIds( false, true ), "Missing line ID langs" );
+		static::assertNotEmpty( $this->transkribusEngine->getValidLineIds( true, false ), "Missing line IDs" );
```
No `true, true` case?

samwilson reviewed (commented):
Looks good! Just one question about duplication of code.

kolakachi committed:
Refactor EngineBaseTest

created a method to instatiate Engines,
refactored dataProvider provideLangs and modified
testFilterValidTranskribusLangs to testFilterValidLangs

Bug: T343003

kolakachi committed:
Refactor EngineBaseTest

created a method to instatiate Engines,
refactored dataProvider provideLangs and modified
testFilterValidTranskribusLangs to testFilterValidLangs

Bug: T343003

kolakachi committed:
remove comments

kolakachi committed:
Fix liniting errors

kolakachi committed:
update instatiateEngine method

kolakachi committed:
Merge branch 'main' into add-test-cases-for-transkribus-related-methods

kolakachi committed:
update Transkribus instatiation

samwilson commented on tests/Engine/EngineBaseTest.php:
```diff
@@ -89,41 +83,48 @@ public function testFilterValidLangsGoogleEngine(): void {
-			// 'equ' is excluded on purpose: T284827
-			'intentionally excluded' => [ [ 'equ' ], [] ],
+	public function provideLangs(): array {
+		return [
```
This is testing only two cases; the old version was testing 3 base cases and each with and without an exception. The new version should be testing those 6 for each engine, for a total of 12 tests. We can fix that in a follow-up though.
